### DISCUSSION TITLE
Error on Startup whili initializing Keyboard layouts

### DISCUSSION THREAD

**Discussion by AlanAndrey** [SafeExamBrowser/seb-win-refactoring#1032] [Q&A]
I encountered an error on startup and was wondering if anybody knows a fix? As far as I can tell from the log files, the error occurs during the initialization of keyboard layouts.

[2024-11-21_21h11m58s_Client.log](https://github.com/user-attachments/files/17851568/2024-11-21_21h11m58s_Client.log)
[2024-11-21_21h11m58s_Runtime.log](https://github.com/user-attachments/files/17851569/2024-11-21_21h11m58s_Runtime.log)

Thanks in advance!

**Comment by dbuechel** [comment]
There indeed appears to be an issue with the language settings as the log reveals:

```
2024-11-21 21:12:07.211 [01] - INFO: [Keyboard] Detected keyboard layout 'Deutsch (Schweiz)' [DEU, Deutsch (Schweiz)].
2024-11-21 21:12:07.211 [01] - INFO: [Keyboard] Detected keyboard layout 'Deutsch (Deutschland)' [DEU, Deutsch (Schweiz)].
2024-11-21 21:12:07.226 [01] - ERROR: Caught unexpected exception while performing operation 'LazyInitializationOperation'!

   Exception Message: Die Kultur wird nicht unterstützt.
Parametername: culture
3072 (0x0c00) ist ein ungültiger Kulturbezeichner.
   Exception Type: System.Globalization.CultureNotFoundException

   bei System.Globalization.CultureInfo.InitializeFromCultureId(Int32 culture, Boolean useUserOverride)
   bei System.Windows.Forms.InputLanguage.get_Culture()
   bei SafeExamBrowser.SystemComponents.Keyboard.Keyboard.Initialize() in C:\Users\appveyor\projects\seb-win-refactoring-t7rrnbsqeloke9rp\SafeExamBrowser.SystemComponents\Keyboard\Keyboard.cs:Zeile 54.
   bei SafeExamBrowser.Client.Operations.ShellOperation.InitializeSystemComponents() in C:\Users\appveyor\projects\seb-win-refactoring-t7rrnbsqeloke9rp\SafeExamBrowser.Client\Operations\ShellOperation.cs:Zeile 229.
   bei SafeExamBrowser.Client.Operations.ShellOperation.Perform() in C:\Users\appveyor\projects\seb-win-refactoring-t7rrnbsqeloke9rp\SafeExamBrowser.Client\Operations\ShellOperation.cs:Zeile 85.
   bei SafeExamBrowser.Core.OperationModel.OperationSequence.Perform() in C:\Users\appveyor\projects\seb-win-refactoring-t7rrnbsqeloke9rp\SafeExamBrowser.Core\OperationModel\OperationSequence.cs:Zeile 108.
```

Please try to remove your currently configured languages, restart your system and then re-add your preferred languages via the system control panel.

**Reply by AlanAndrey** [reply]
Thank you for your quick response. Uninstalling the languages sadly didn't work.
[2024-12-03_12h33m04s_Client.log](https://github.com/user-attachments/files/17992659/2024-12-03_12h33m04s_Client.log)
[2024-12-03_12h33m04s_Runtime.log](https://github.com/user-attachments/files/17992661/2024-12-03_12h33m04s_Runtime.log)

Do you have any idea what else could cause this error?

**Reply by dbuechel** [reply]
Could you please provide a screen shot of the languages you've configured in the system itself?

**Comment by AlanAndrey** [comment]
Shure:
![WhatsApp Image 2024-12-07 at 10 57 45_827c3357](https://github.com/user-attachments/assets/e8bba467-e3fe-49ae-ac8f-a52dd7d9a371)

**Reply by dbuechel** [reply]
Thanks for the update. I have installed the same languages and cannot reproduce the issue. The only explanation I currently have is that there is an issue with your system (configuration) or .NET installation. As next steps, you should make sure that your system is up to date (installing all pending updates), potentially checking and repairing all system files (using the System File Checker `sfc.exe`) and then try to reinstall the .NET Framework.
